docs/schemaform/form-config.md

```diff
@@ -0,0 +1,297 @@
+  }
+}
+```
+
+The `schema` and `uiSchema` objects should have similar structure. In other words, they should have the same fields organized the same way. The main difference between the structure of the two objects is that the uiSchema object does not have to contain all the fields that the schema object does and it does not need a `properties` object for sub-fields. So given this schema, for example:
+
+```js
+{
+  type: 'object',
+  properties: {
+    field1: {
+      type: 'string'
+    }
+  }
+}
+```
+
+The matching uiSchema would be:
+
+```js
+{
+  'ui:title': 'My form',
+  field1: {
+    'ui:title': 'My field'
+  }
+}
+```
+
+This does not apply to array fields; for those, you still need to specify an `items` object that contains the fields for each row in the array.
+
+### uiSchema configuration
+In addition to the uiSchema options listed in the library docs, we have some additional options that are supported for all forms:
+
+```js
+{
+  // We use this instead of the title property in the JSON Schema
+  'ui:title': '', 
+  
+  // We use this instead of the description property in the JSON Schema. This can be
+  // a string or a React component and would normally used on object fields in the
+  // schema to provide description text or html before a block of fields
+  'ui:description': '' || DescriptionComponent,
+  
+  // Customize the field or widget you're using
+  'ui:field': '' || FieldComponent,
+  'ui:widget': '' || WidgetComponent,
+  
+  // This widget will be shown on the review page. Should always be used if you specify
+  // a custom widget component, but can be used with regular widgets as well. Currently
+  // only implemented for string fields
+  'ui:reviewWidget': WidgetComponent,
+
+  // Use this to provide a function to make a field conditionally required. First
+  // argument is the current form data and the second is the formContext object,
+  // which will contain the form data for other pages (tbd). You should avoid making
+  // a field required in the JSON schema and using `ui:required` on the same field.
+  'ui:required': function (pageData) {
+    return true || false;
+  },
+  
+  // This is an array of validation functions or objects that can be used to add validation
+  // that is not possible through JSON Schema. See below for the properties passed
+  // to the validation functions and how to use them.
+  'ui:validations': [
+    function (errors, fieldData, pageData, fieldSchema, errorMessages) {
+      errors.addError('My error');
+    },
+    {
+      validator: (errors, fieldData, pageData, fieldSchema, errorMessages, options) => {
+        errors.addError('My other error');
+      },
+      options: {}
+    }
+  ],
+  
+  // An object with field specific error messages. Structured by error name (from JSON
+  // Schema error types). This is passed to custom validations in `ui:validations` if
+  // you want to allow configurable error messages in a validator.
+  'ui:errorMessages': {
+    errorType: ''
```

Comment:
Might be useful to enter a sample error message so it's more clear how this would be used?